src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -169,14 +171,114 @@ private function createRelationsForItem($item, $formattedRelations)
                 }
             } elseif ($relation instanceof HasOne || $relation instanceof MorphOne) {
                 $modelInstance = $relation->updateOrCreate([], $relationDetails['values']);
+            } elseif ($relation instanceof HasMany || $relation instanceof MorphMany) {
+                $relation_values = $relationDetails['values'][$relationMethod];
+                // if relation values are null we can only attach, also we check if we sent a single dimensional array [1,2,3], or an array of arrays: [[1][2][3]]
+                // if is as single dimensional array we can only attach.
+                if ($relation_values === null || count($relation_values) == count($relation_values, COUNT_RECURSIVE)) {
+                    $this->attachManyRelation($item, $relation, $relationDetails, $relation_values);
+                } else {
+                    $this->createManyEntries($item, $relation, $relationMethod, $relationDetails);
+                }
             }
-
             if (isset($relationDetails['relations'])) {
                 $this->createRelationsForItem($modelInstance, ['relations' => $relationDetails['relations']]);
             }
         }
     }
 
+    /**
+     * When using the HasMany/MorphMany relations as selectable elements we use this function to "mimic-sync" in those relations.
+     * Since HasMany/MorphMany does not have the `sync` method, we manually re-create it.
+     * Here we add the entries that developer added and remove the ones that are not in the list.
+     * This removal process happens with the following rules:
+     * - by default Backpack will behave like a `sync` from M-M relations: it deletes previous entries and add only the current ones.
+     * - `force_delete` is configurable in the field, it's `true` by default. When false, if connecting column is nullable instead of deleting the row we set the column to null.
+     * - `fallback_id` could be provided. In this case instead of deleting we set the connecting key to whatever developer gives us.
+     *
+     * @return void
+     */
+    public function attachManyRelation($item, $relation, $relationDetails, $relation_values)
```

Comment:
```suggestion
    private function attachManyRelation($item, $relation, $relationDetails, $relation_values)
```

From what I see, this method is only called inside `createRelationsForItem()`, which itself is on the CrudPanel object and `private`...

Any benefit to having this `public` @pxpm ?

Reply:
I'm pretty sure we're going to refractor this in the next 12 months, so it'd be better to only add `private` methods, so we can do that without a breaking change.